### PR TITLE
[11.8] Allow specifying params for Wiki::showAll()

### PR DESCRIPTION
**Pull request by spawnia**
https://docs.gitlab.com/ee/api/wikis.html#list-wiki-pages

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
You should use the result pager to fetch different pages, assuming that is your use case here.

**Comment by spawnia** [comment]
Thanks for the hint, I will definitely keep that in mind as we probably have some wikis with over 20 pages 👍

**Comment by spawnia** [comment]
Passing the option is still useful though:

```php
        $pages = (new ResultPager($this->gitlab->connection()))->fetchAll(
            $this->gitlab->wiki(),
            'showAll',
            [
                $projectId,
                ['with_contents' => true]
            ],
        );
```

**Comment by spawnia** [comment]
Is there anything else I can do here? Would love to see this merged, I am currently blocked by being unable to use the GitLab API to its full extent.

**Review by 20uf — APPROVED** [review]

File: src/Api/Wiki.php [hunk 1]
```diff
@@ -17,8 +17,8 @@
 class Wiki extends AbstractApi
 {
     /**
-     * @param int|string $project_id
-     * @param array      $params
+     * @param int|string          $project_id
+     * @param array<string,mixed> $params
```
Comment (GrahamCampbell) [inline src/Api/Wiki.php:21]: Please adjust this doc to be similar to the other parameters doc, and use the options resolver.

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: src/Api/Wiki.php (hunk 1 above)
Comment (spawnia) [reply src/Api/Wiki.php:21]: Done.

**Comment by spawnia** [comment]
I don't know what to do about the [test failure](https://github.com/GitLabPHP/Client/runs/4642485219?check_suite_focus=true), tests against the local PHP 7.4 setup (using `make`) work.

**Comment by GrahamCampbell** [comment]
I was unable to push to your branch because you unchecked the option. I'm going to re-create this PR.

**Comment by GrahamCampbell** [comment]
Thanks for the PR. :)

**Pull request closed without merging**